core/metacling/src/TCling.cxx

```diff
@@ -4288,6 +4288,8 @@ TCling::CheckClassInfo(const char *name, Bool_t autoload, Bool_t isClassOrNamesp
 
 Bool_t TCling::CheckClassTemplate(const char *name)
 {
+   // Interpreter transaction ahead, needs locking
+   R__LOCKGUARD(gInterpreterMutex);
```

Comment:
Perhaps I am wrong but it seems that getting the lookup helper is a thread safe operation, being cling::getLookupHelper a const method, cling being well written and the lh being instantiated[ in the constructor of the interpreter](https://github.com/root-project/cling/blob/45318c5a0ed29bd32639531bd0fa06d6d3f4d20b/lib/Interpreter/Interpreter.cpp#L254). Perhaps the guard can be put after getting the lh? It won't make the section drastically smaller, but still...

Reply:
Indeed, the non-thread-safe operation is in `findClassTemplate` so I can move the lock a bit further down :+1: